test/unit/org/apache/cassandra/service/reads/DigestResolverTest.java

```diff
@@ -78,6 +79,69 @@ public void noRepairNeeded()
         assertPartitionsEqual(filter(iter(response)), resolver.getData());
     }
 
+    @Test
+    public void multiThreadedNoRepairNeededReadCallback() throws InterruptedException
+    {
+        SinglePartitionReadCommand command = SinglePartitionReadCommand.fullPartitionRead(cfm, nowInSec, dk);
+        EndpointsForToken targetReplicas = EndpointsForToken.of(dk.getToken(), full(EP1), full(EP2));
+        PartitionUpdate response = update(row(1000, 4, 4), row(1000, 5, 5)).build();
+        ReplicaPlan.SharedForTokenRead plan = plan(ConsistencyLevel.ONE, targetReplicas);
+
+        ExecutorService pool = Executors.newFixedThreadPool(4);
+
+        try
+        {
+            for (int i = 0; i < 10_000_000; i++)
```

Comment:
This test can obviously run for quite a while, but if it doesn't run long enough to catch the problem, it doesn't add much value. I'm open to throwing an `@Ignore` on it for now, leaving it to a suite of tests we don't run on every commit, or any other ideas...

Reply:
I think that the time bound of 1/2 of the JUnit timeout is a nice solution. If we find it problematic in the future we can always move it to long tests.